src/sentry/api/bases/organization.py

```diff
@@ -385,14 +386,24 @@ def _filter_projects_by_permissions(
             if force_global_perms:
                 span.set_tag("mode", "force_global_perms")
             else:
-                if (
-                    user
-                    and is_active_superuser(request)  # superuser should fetch all projects
-                    or not filter_by_membership  # explicitly requested projects
-                    or include_all_accessible  # requested $all projects
-                ):
+                active_superuser = user and is_active_superuser(request)
+                # Superuser should fetch all projects.
+                # Also check if explicitly requesting all projects with include_all_accessible
+                if active_superuser or include_all_accessible:
                     span.set_tag("mode", "has_project_access")
                     func = request.access.has_project_access
+                # Check if explicitly requesting specific projects
+                elif not filter_by_membership:
+                    if is_active_staff(request):
+                        # There is a special case for staff, where we want to fetch usage stats for a single
+                        # project in _admin using OrganizationStatsEndpointV2 but cannot use as_project_access
+                        # like superuser because it fails. The workaround is to create a lambda that mimics
+                        # checking for active projects like has_project_access without further validation.
+                        span.set_tag("mode", "staff_fetch_all")
+                        func = lambda proj: proj.status == ObjectStatus.ACTIVE  # noqa: E731
```

Comment:
also this `get_projects` function is accessible from every single endpoint that subclasses `OrganizationEndpoint` (or some other class that also subclasses `OrganizationEndpoint`, etc). but i don't think any of the other endpoints that call it allow staff

Reply:
_admin fetches stats for any project for event categories (error, transaction, etc.) with all statuses (filtered, dropped, etc.) when you go that project's page.

Reply:
I can assert `len(projects) == 1` if we hit that staff check

Reply:
actually, i think it's okay to not add an assertion

Reply:
I can add it to the if clause like `elif not filter_by_membership and len(projects == 1):

Reply:
if you can hit the endpoint for a single project for all projects i don't see why we'd need to prevent people from hitting multiple projects at once, just ignore me lol